app/controllers/members_controller.rb

```diff
@@ -1,74 +1,79 @@
+# frozen_string_literal: true
+
+# Members Controller
 class MembersController < ApplicationController
-  before_action :set_member, only: [:show, :edit, :update, :destroy]
+  before_action :set_member, only: %i[show edit update destroy]
 
   # GET /members
-  # GET /members.json
   def index
-    @members = Member.all
+    @members =
+      if search_query
+        Member.search_result(search_query)
+      else
+        Member.all
+      end
   end
 
   # GET /members/1
   # GET /members/1.json
-  def show
-  end
+  def show; end
 
   # GET /members/new
   def new
     @member = Member.new
   end
 
   # GET /members/1/edit
-  def edit
-  end
+  def edit; end
 
   # POST /members
-  # POST /members.json
   def create
     @member = Member.new(member_params)
 
     respond_to do |format|
       if @member.save
-        format.html { redirect_to @member, notice: 'Member was successfully created.' }
-        format.json { render :show, status: :created, location: @member }
+        format.html { redirect_to @member, notice: 'Member was created.' }
       else
         format.html { render :new }
-        format.json { render json: @member.errors, status: :unprocessable_entity }
       end
     end
   end
 
   # PATCH/PUT /members/1
-  # PATCH/PUT /members/1.json
   def update
     respond_to do |format|
       if @member.update(member_params)
-        format.html { redirect_to @member, notice: 'Member was successfully updated.' }
+        format.html { redirect_to @member, notice: 'Member was updated.' }
         format.json { render :show, status: :ok, location: @member }
       else
         format.html { render :edit }
-        format.json { render json: @member.errors, status: :unprocessable_entity }
       end
     end
   end
 
   # DELETE /members/1
-  # DELETE /members/1.json
   def destroy
     @member.destroy
     respond_to do |format|
-      format.html { redirect_to members_url, notice: 'Member was successfully destroyed.' }
-      format.json { head :no_content }
+      format.html { redirect_to members_url, notice: 'Member was destroyed.' }
     end
   end
 
   private
-    # Use callbacks to share common setup or constraints between actions.
-    def set_member
-      @member = Member.find(params[:id])
-    end
 
-    # Never trust parameters from the scary internet, only allow the white list through.
-    def member_params
-      params.fetch(:member, {}).permit(:username, :firstname, :lastname, :email, :bio)
-    end
+  # Use callbacks to share common setup or constraints between actions.
+  def set_member
+    @member = Member.find(params[:id])
+  end
+
+  # Never trust parameters; allow only white listed attributes through.
+  def member_params
+    params.fetch(:member, {})
+          .permit(:username, :firstname, :lastname, :email, :bio, :query)
+  end
+
+  # Don't trust the parameters from the search form either.
+  def search_query
+    params[:query]
```

Comment:
Use `member_params[:query]` here to ensure you are using the _strong_ permitted version.